shopfloor_reception_packaging_dimension/services/reception.py

```diff
@@ -0,0 +1,234 @@
+
+        """
+        for dimension in dimensions:
+            if local_table[dimension] is None:
+                if dimension in self.dimension_to_update.keys():
+                    self.dimension_to_update.pop(dimension)
+            else:
+                self.dimension_to_update[dimension] = local_table[dimension]
+
+    def set_packaging_dimension(
+        self,
+        picking_id,
+        selected_line_id,
+        packaging_id,
+        height=None,
+        length=None,
+        width=None,
+        weight=None,
+        qty=None,
+        barcode=None,
+        cancel=False,
+    ):
+        """Set the dimension on a product packaging.
+
+        If the user cancel the dimension update we still propose the next
+        possible packgaging.
+
+        Transitions:
+            - set_packaging_dimension: if more packaging needs dimension
+            - set_quantity: otherwise
+        """
+        self._store_dimension_to_update(
+            ["height", "length", "width", "weight", "barcode", "qty"], locals()
+        )
+        picking = self.env["stock.picking"].browse(picking_id)
+        selected_line = self.env["stock.move.line"].browse(selected_line_id)
+        packaging = self.env["product.packaging"].sudo().browse(packaging_id)
+        message = None
+        next_packaging = None
+        if not packaging:
+            message = self.msg_store.record_not_found()
+        elif not cancel and self._check_dimension_to_update():
+            self._update_packaging_dimension(packaging)
+            message = self.msg_store.packaging_dimension_updated(packaging)
+        if packaging:
+            next_packaging = self._get_next_packaging_to_set_dimension(
+                selected_line.product_id, packaging
+            )
+        if not next_packaging:
+            return self._response_for_set_quantity(
+                picking, selected_line, message=message
+            )
+        return self._response_for_set_packaging_dimension(
+            picking, selected_line, next_packaging, message=message
+        )
+
+    def _check_dimension_to_update(self):
+        """Return True if there is any dimension that needs to be updated on the packaging."""
+        dimension_to_check = ["height", "length", "width", "weight", "barcode", "qty"]
+        return any(
+            [self.dimension_to_update.get(dim, False) for dim in dimension_to_check]
+        )
+
+    def _update_packaging_dimension(self, packaging):
+        """Update dimension on the packaging."""
+        dimension_to_update = ["height", "length", "width", "weight", "barcode", "qty"]
+        for dimension in dimension_to_update:
+            value = self.dimension_to_update.get(dimension, None)
+            if value is not None:
+                if dimension == "weight":
+                    dimension = "max_weight"
+                elif dimension == "length":
+                    dimension = "packaging_length"
+                packaging[dimension] = value
```

Comment:
Please do not repeat the fields. 

```suggestion
    def _get_packaging_fields_to_update(self):
        return ["height", "length", "width", "weight", "barcode", "qty"]
    
    def _check_dimension_to_update(self):
        """Return True if there is any dimension that needs to be updated on the packaging."""
        dimension_to_check = self._get_packaging_fields_to_update()
        return any(
            [self.dimension_to_update.get(dim, False) for dim in dimension_to_check]
        )

    def _update_packaging_dimension(self, packaging):
        """Update dimension on the packaging."""
        dimension_to_update = self._get_packaging_fields_to_update()
        for dimension in dimension_to_update:
            value = self.dimension_to_update.get(dimension, None)
            if value is not None:
                if dimension == "weight":
                    dimension = "max_weight"
                elif dimension == "length":
                    dimension = "packaging_length"
                packaging[dimension] = value
```

Reply:
Updated